src/rules/javascript/eval_string_arg.js

```diff
@@ -0,0 +1,31 @@
+import { EVAL_STRING_ARG } from 'messages/javascript';
+import { getVariable } from 'utils';
+
+export default function(context) {
+  return {
+    CallExpression: function(node) {
+      var acceptedArgs = ['FunctionExpression', 'ArrowFunctionExpression'];
+
+      if (node.callee.property.type === 'Identifier' &&
+          (node.callee.property.name === 'setTimeout' ||
+           node.callee.property.name === 'setInterval')) {
+
+        if (node.arguments.length > 0 &&
+            node.arguments[0].type === 'Identifier') {
+          var originalDef = getVariable(context, node.arguments[0].name);
```

Comment:
Let's add some comments to this explaining what's going on. It will help massively in the future. We should all comment more in the rules.